### PR TITLE
Rabbitmq volttron auth

### PR DESCRIPTION
**Pull request by shwethanidd**
# Description
This PR contains 
Documentation updates to README
More pytests for rmq pubsub 
Issue #1873: To fix unwanted log messages appearing on the console when run with 
volttron -vv -l volttron.log&
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/schandrika%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23issuecomment-451303114%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222019-01-03T22%3A50%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20ad0a52ecdbf6c6c27d65eecbed5ed6c2d08cb59c%20volttron/platform/control.py%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23discussion_r245106194%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%27list-exchanges%27%20has%20help%20text%20of%20%27add%27%20a%20new%20exchange%3F%22%2C%20%22created_at%22%3A%20%222019-01-03T19%3A24%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222019-01-03T22%3A50%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttron/platform/control.py%3AL2412-2419%22%7D%2C%20%22Pull%20ad0a52ecdbf6c6c27d65eecbed5ed6c2d08cb59c%20README.md%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23discussion_r245106845%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22It%20would%20be%20good%20to%20mention%20in%20one%20of%20the%20initial%20rabbitmq%20documentation%20that%20this%20is%20just%20a%20summary/overview%20written%20based%20on%20rabbitmq%20official%20documentation%20and%20that%20images%20are%20also%20from%20rabbitmq%20docs.%22%2C%20%22created_at%22%3A%20%222019-01-03T19%3A26%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%2C%20%7B%22body%22%3A%20%22Added%20a%20note%20in%20the%20beginning%20of%20docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222019-01-03T22%3A48%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/19616893%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shwethanidd%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222019-01-03T22%3A50%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20README.md%3AL486-502%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23issuecomment-451303114%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23discussion_r245106194%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23discussion_r245106845%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23discussion_r245158558%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23discussion_r245159013%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1874%23discussion_r245159016%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/schandrika'></a>

- [x] <a href='#crh-comment-Pull ad0a52ecdbf6c6c27d65eecbed5ed6c2d08cb59c volttron/platform/control.py 5'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1874#discussion_r245106194'>File: volttron/platform/control.py:L2412-2419</a></b>
- <a href='https://github.com/schandrika'></a> 'list-exchanges' has help text of 'add' a new exchange?
- [x] <a href='#crh-comment-Pull ad0a52ecdbf6c6c27d65eecbed5ed6c2d08cb59c README.md 5'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1874#discussion_r245106845'>File: README.md:L486-502</a></b>
- <a href='https://github.com/schandrika'></a> It would be good to mention in one of the initial rabbitmq documentation that this is just a summary/overview written based on rabbitmq official documentation and that images are also from rabbitmq docs.
- <a href='https://github.com/shwethanidd'></a> Added a note in the beginning of docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1874?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1874?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1874?approve=0'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1874'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by craig8** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1874'>Refresh</a>.<a href='#crh-update-none'></a>

File: README.md [hunk 1]
```diff
@@ -486,15 +486,16 @@ upstream servers on the downstream server and make the VOLTTRON exchange
 
 **Using Shovel Plugin**
 
-In RabbitMQ based VOLTTRON, forwarder and data mover agents will be replaced by shovels
+In RabbitMQ based VOLTTRON, forwarder and data puller agents will be replaced by shovels
```
[the remark this answers is not in the document]
Comment (shwethanidd) [reply README.md]: Added a note in the beginning of docs/source/core_services/messagebus_refactor/RabbitMQ-Overview.rst

File: volttron/platform/control.py [hunk 2]
```diff
@@ -2412,7 +2412,7 @@ def add_parser(*args, **kwargs):
         rabbitmq_list_user_properties.add_argument('user', help='RabbitMQ user id')
         rabbitmq_list_user_properties.set_defaults(func=list_user_properties)
 
-        rabbitmq_list_exchanges = add_parser('list-exchanges', help='add a new user',
+        rabbitmq_list_exchanges = add_parser('list-exchanges', help='add a new exchange',
```
Comment (schandrika) [inline volttron/platform/control.py]: :+1:<a href='#crh-update-none'></a>

File: README.md (hunk 1 above)
Comment (schandrika) [inline README.md]: :+1:<a href='#crh-update-complete'></a>

**Comment by schandrika** [comment]
:shipit:

**Pull request merged**